### PR TITLE
Add feature for async media image

### PR DESCRIPTION
**Pull request by zhongwuzw**
Todo:
- [ ] Add entry in `CHANGELOG`.

### CODE REVIEW

File: Sources/Views/Cells/MediaMessageCell.swift [hunk 1]
```diff
@@ -64,5 +69,7 @@ open class MediaMessageCell: MessageCollectionViewCell {
         default:
             break
         }
+        
+        displayDelegate.configureMediaMessageImageView(imageView, for: message, at: indexPath, in: messagesCollectionView)
```
Comment (SD10) [inline Sources/Views/Cells/MediaMessageCell.swift:73]: Interesting approach. Let me think about this some more. My concern is this then overwriting the `image` in the `.photo` enum. 🤔

**Comment by Jeyhey** [comment]
@SD10 Well, the proposed approach in the pull request is similar to how the video enum is defined. In that sense it is consistent. Could be a good approach after all.

**Review by nathantannar4 — APPROVED** [review]

**Comment by SD10** [comment]
So after we add something like the `PhotoItem` protocol from #582 we will get something very similar to this existing enum case. My question is this:

Do we want to add the following method as a hook to add images async?
```Swift
func configureMediaMessageImageView(_ imageView: UIImageView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView)
```
Or should we make this more generic and just give users access to the `MessageContainerView`?
```Swift
func configureMessageContainerView(_container: MessageContainerView, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView)
```
The reason being maybe users have `.custom` data types that need async support.

Let's also consider that users can achieve this without a method at all, simply by overriding `cellForItem(at indexPath:)`.

cc @zhongwuzw

**Comment by zhongwuzw** [comment]
@SD10 , I don't like overriding `cellForItem` method, because user needs to judge kind of `cell` and call `super` ...
I still prefer to choose provide `imageView`, If We only provide `containerView`, users maybe cannot set image easily.

**Comment by SD10** [comment]
Going to be too difficult to resolve the conflicts for this one but I'll open an issue

**Pull request closed without merging**